pod/annotations.go

```diff
@@ -93,6 +93,10 @@ const (
 	AnnotationKeyPodPriorityClassIntent     = "pod.titus.netflix.com/priority-class-intent"
 	AnnotationKeyPodScheduledInTrough       = "pod.titus.netflix.com/scheduled-in-trough"
 	AnnotationKeyPodPreemptionResubmitCount = "resubmit-number.pod.netflix.com/preemption"
+	// AnnotationKeyPodTerminationReason is a human readable string indicating *why* a pod was terminated.
+	// It is the responsibility of any code that calls pod.delete() to populate this annotation so that operators
+	// of the system can understand why a pod was deleted.
+	AnnotationKeyPodTerminationReason = "pod.titus.netflix.com/pod-termination-reason"
```

Comment:
Can we add:
* `pod.titus.netflix.com/pod-termination-reason-code` with a set of reasonable reason codes (we could map it more easily to Titus task reason code)
* `pod.titus.netflix.com/pod-termination-by-caller` (this could be a specific controller making the termination request, or example `TitusJobCoordinator/nodeGcController`

Reply:
> Can we add:
> 
> * `pod.titus.netflix.com/pod-termination-reason-code` with a set of reasonable reason codes (we could map it more easily to Titus task reason code)

This is a different type of thing and already encoded in the pod status.
In a sense, it will always be "KillInitiated", right?


> * `pod.titus.netflix.com/pod-termination-by-caller` (this could be a specific controller making the termination request, for example `TitusJobCoordinator/nodeGcController`

Sounds fine

Reply:
`KillInitiated` is a state. But there is also a `ReasonCode` associated with each state providing more insight.

Reply:
I see. I've added the annotation key and the values that could be used